From: Henryk Heisig <hyniu@o2.pl>
Date: Thu, 29 Jun 2017 15:20:31 +0200
Subject: ar71xx: Archer C58/C59/C60 fix qca9886 wireless interface

This commit fix 5GHz wireless interface used in Archer C58/C59/C60v1
and set correctly MAC address on this interface.

Signed-off-by: Henryk Heisig <hyniu@o2.pl>

diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
index 04d8bcca535fd341aa0af29cafc09c4f20980c3a..669b8d3387702201270d5a889ee7c23f5c925ff4 100644
--- a/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
+++ b/target/linux/ar71xx/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
@@ -132,7 +132,9 @@ case "$FIRMWARE" in
 	;;
 "ath10k/pre-cal-pci-0000:00:00.0.bin")
 	case $board in
-	archer-c58-v1)
+	archer-c58-v1|\
+	archer-c59-v1|\
+	archer-c60-v1)
 		ath10kcal_extract "art" 20480 12064
 		;;
 	esac
diff --git a/target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac b/target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
index 7d2eca546d76b771b12026788510f73a293a9a93..669b208231e43fe86e998c7202c133c86ae0bf8d 100644
--- a/target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
+++ b/target/linux/ar71xx/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac
@@ -12,7 +12,9 @@ PHYNBR=${DEVPATH##*/phy}
 board=$(ar71xx_board_name)
 
 case "$board" in
-	archer-c58-v1)
+	archer-c58-v1|\
+	archer-c59-v1|\
+	archer-c60-v1)
 		echo $(macaddr_add $(mtd_get_mac_binary mac 8)  $(($PHYNBR - 1)) ) > /sys${DEVPATH}/macaddress
 		;;
 	*)
diff --git a/target/linux/ar71xx/image/tp-link.mk b/target/linux/ar71xx/image/tp-link.mk
index d639b3e0ddb8dfd68aca6cd0305da3559fbec1a8..e6c96fd89abcc3d78a58ef857ce12d82bc937be8 100644
--- a/target/linux/ar71xx/image/tp-link.mk
+++ b/target/linux/ar71xx/image/tp-link.mk
@@ -121,7 +121,7 @@ TARGET_DEVICES += archer-c25-v1
 
 define Device/archer-c58-v1
   DEVICE_TITLE := TP-LINK Archer C58 v1
-  DEVICE_PACKAGES := kmod-ath10k
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca9888
   BOARDNAME := ARCHER-C58-V1
   TPLINK_BOARD_NAME := ARCHER-C58-V1
   DEVICE_PROFILE := ARCHERC58V1
@@ -136,7 +136,7 @@ TARGET_DEVICES += archer-c58-v1
 
 define Device/archer-c59-v1
   DEVICE_TITLE := TP-LINK Archer C59 v1
-  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport kmod-ath10k
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport kmod-ath10k ath10k-firmware-qca9888
   BOARDNAME := ARCHER-C59-V1
   TPLINK_BOARD_NAME := ARCHER-C59-V1
   DEVICE_PROFILE := ARCHERC59V1
@@ -151,7 +151,7 @@ TARGET_DEVICES += archer-c59-v1
 
 define Device/archer-c60-v1
   DEVICE_TITLE := TP-LINK Archer C60 v1
-  DEVICE_PACKAGES := kmod-ath10k
+  DEVICE_PACKAGES := kmod-ath10k ath10k-firmware-qca9888
   BOARDNAME := ARCHER-C60-V1
   TPLINK_BOARD_NAME := ARCHER-C60-V1
   DEVICE_PROFILE := ARCHERC60V1
